sources/common/inc/LogParser.h

```diff
@@ -61,6 +69,17 @@ namespace PARSER
                 return std::move(m_pSignalDb);
             }
 
+            /// Invokes the parser.
+            void Execute();
+
+            /// Attaches an instrument to the parser.
+            ///
+            /// @param instrument An instrument to be attached.
+            void Attach(INSTRUMENT::Instrument &instrument);
```

Comment:
We weren't doing so but I like the idea. I will update this at least for the files I will be touching in this  issue.